decoder/stats.go

```diff
@@ -375,6 +422,154 @@ func updatePokemonStats(old *Pokemon, new *Pokemon, areas []geo.AreaName) {
 	}
 }
 
+func updateRaidStats(old *Gym, new *Gym, areas []geo.AreaName) {
+	if len(areas) == 0 {
+		areas = []geo.AreaName{
+			{
+				Parent: "world",
+				Name:   "world",
+			},
+		}
+	}
+
+	locked := false
+
+	// Loop though all areas
+	for i := 0; i < len(areas); i++ {
+		area := areas[i]
+
+		// Check if Raid has started/is active or RaidEndTimestamp has changed (Back-to-back raids)
+		if new.RaidPokemonId.ValueOrZero() > 0 &&
+			(old.RaidPokemonId != new.RaidPokemonId || old.RaidEndTimestamp != new.RaidEndTimestamp) {
```

Comment:
```suggestion
		(old == nil || old.RaidPokemonId != new.RaidPokemonId || old.RaidEndTimestamp != new.RaidEndTimestamp) {
```
you need the old nil check, otherwise this can panic

Reply:
Great catch! 🤦